### ISSUE TITLE
CMake 3.14: FetchContent instead of ExternalProject

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#11123]
suggested by @slurps-mad-rips https://github.com/libuv/libuv/pull/2373#issuecomment-536349553

> CMake 3.14 as this has a configure time module called `FetchContent`, which is a superior version of `ExternalProject`, and allows "bundling" dependencies, while also supporting git worktree checkouts so you can even patch local forks or use git clones of alternative distributions.

**Comment by RenFraser** [comment]
Is anyone currently working on this?

**Comment by RenFraser** [comment]
I've had a quick look at the third party deps. Looks like This is going to be a pretty big undertaking. I've done it before and it'll be a much more manageable task to break up the work into smaller increments.

**Comment by jamessan** [comment]
> Is anyone currently working on this?

No, and I don't think there's much reason to right now.  We're still supporting CMake 3.10.

**Comment by dundargoc** [comment]
Superseded by https://github.com/neovim/neovim/issues/24004
